cmd/openshift-install/create.go

```diff
@@ -290,53 +292,55 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
 
 	eventTimeout := 30 * time.Minute
 	logrus.Infof("Waiting up to %v for the bootstrap-complete event...", eventTimeout)
-	eventContext, cancel := context.WithTimeout(ctx, eventTimeout)
+	if err := waitForEvent(ctx, events, "bootstrap-complete", eventTimeout); err != nil {
+		return err
+	}
+
+	logrus.Info("Destroying the bootstrap resources...")
+	return destroybootstrap.Destroy(rootOpts.dir)
+}
+
+// waitForEvent watches the events, waits for the event of the given name and prints out all other events on the way.
+func waitForEvent(ctx context.Context, client corev1client.EventInterface, name string, timeout time.Duration) error {
+	waitCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
-	_, err = Until(
-		eventContext,
-		"",
-		func(sinceResourceVersion string) (watch.Interface, error) {
-			for {
-				watcher, err := events.Watch(metav1.ListOptions{
-					ResourceVersion: sinceResourceVersion,
-				})
-				if err == nil {
-					return watcher, nil
-				}
-				select {
-				case <-eventContext.Done():
-					return watcher, err
-				default:
-					logrus.Warningf("Failed to connect events watcher: %s", err)
-					time.Sleep(2 * time.Second)
-				}
+
+	_, err := clientwatch.UntilWithSync(
+		waitCtx,
+		&cache.ListWatch{ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			ret, err := client.List(options)
+			if err != nil {
+				logrus.Warningf("Failed to list events: %s", err)
 			}
-		},
-		func(watchEvent watch.Event) (bool, error) {
-			event, ok := watchEvent.Object.(*corev1.Event)
-			if !ok {
-				return false, nil
+			return ret, err
+		}, WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			ret, err := client.Watch(options)
+			if err != nil {
+				logrus.Warningf("Failed to watch events: %s", err)
 			}
-
-			if watchEvent.Type == watch.Error {
-				logrus.Debugf("error %s: %s", event.Name, event.Message)
+			return ret, err
+		}},
+		&corev1.Event{},
+		nil,
+		func(event watch.Event) (bool, error) {
+			ev, ok := event.Object.(*corev1.Event)
+			if !ok {
+				logrus.Warnf("Expected an core/v1.Event object but got a %q object instead", event.Object.GetObjectKind().GroupVersionKind())
```

Comment:
Is this a transient error?

Reply:
if this happens, it's a programming error and something bad happens. Would leave it as warning.